### ISSUE TITLE
The official release of palettable is broken

### ISSUE THREAD

**Issue by has2k1** [jiffyclub/palettable#47]
**Immediate Problem**

`palettable` requires `setuptools` as a runtime dependency but it is not listed in the official release. This was fixed in https://github.com/jiffyclub/palettable/commit/b20cdca13647941deaaa2c4b15cdb9c56b234add, was not released.

Up to now, this issue has effectively been masked by at-least one of:

   1. Virtual environments tend to pre-install `setuptools` when they are created. This is not a standard requirement.
   2. Packages that depend on `palettable` may also have included `setuptools` as a runtime dependency.

The "new" packaging & installation standards (e.g [PEP 518](https://peps.python.org/pep-0518/), [PEP 621](https://peps.python.org/pep-0621/)) have now been around long enough that installation tools & other packages have started to strictly adhere to them. That means fewer chances of `setuptools` being installed in the user environment.

**Larger Problem**

`palettable` vendors [ez_setup](https://github.com/ActiveState/ez_setup) as a way to sidestep `setuptools` during installation, yet `setuptools` is a runtime dependency! Juggling build/installation time and runtime dependencies is best done in a standards compliant way.
`ez_setup` predates these standards and may contribute to or mask other installation problems.

**Solution**
*Possible immediate solution*: Make a release.
*Better immediate solution*: Consider PR https://github.com/jiffyclub/palettable/pull/46, remove `setuptools` as a runtime dependency and make a release.
*Final solution*: Move to a standard installation process.

**Comment by jiffyclub** [comment]
I've merged #46 and released version 3.3.1 based on that. I'm a bit out of date on Python packaging (this package was originally made over 10 years ago!) so I'm going to brush up and see about updating to a more recent packaging design.

**Comment by has2k1** [comment]
[pyopensci](https://www.pyopensci.org/) has a comprehensive [python packaging guide](https://www.pyopensci.org/python-package-guide/index.html) with a relevant page on how to use [pyproject.toml](https://www.pyopensci.org/python-package-guide/package-structure-code/pyproject-toml-python-package-metadata.html).

**Comment by jiffyclub** [comment]
I've released version 3.3.3 with `pyproject.toml` based setup, hopefully this takes care of things for you. Thanks for the pointer! For my own future reference these are the pages I found most useful:
- Packaging Flow: https://packaging.python.org/en/latest/flow/
- Project Metadata Descriptions: https://packaging.python.org/en/latest/specifications/declaring-project-metadata/
- Setuptools with pyroject.toml: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

**Comment by kiefernforst** [comment]
You could've kept `setup.py`. The PEP 621 build fails with

```
* Building wheel...
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/usr/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/usr/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
    return _build_backend().build_wheel(wheel_directory, config_settings,
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 436, in build_wheel
    return _build(['bdist_wheel'])
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 427, in _build
    return self._build_with_temp_dir(
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 408, in _build_with_temp_dir
    self.run_setup()
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 317, in run_setup
    exec(code, locals())
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'ez_setup'
```

because ez_setup.py has been killed in https://github.com/pypa/setuptools/commit/a73118b9e3c125910ce3a50ab1e676632b4c12d1
